internal/provider/network/isolated_resource.go

```diff
@@ -0,0 +1,670 @@
+		return
+	}
+
+	// Get Org and VDC
+	org, _, err = r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving Org", err.Error())
+		return
+	}
+
+	// Build URN Network
+	urn := state.ID.ValueString()
+	// Get network
+	orgNetwork, err := org.GetOpenApiOrgVdcNetworkById(urn)
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			// Network not found, so remove from state
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("[READ] Error retrieving isolated network", err.Error())
+		return
+	}
+
+	plan := &networkIsolatedResourceModel{
+		ID:           types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID),
+		OwnerID:      types.StringValue(extractUUID(orgNetwork.OpenApiOrgVdcNetwork.OwnerRef.ID)),
+		Name:         types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Name),
+		Description:  types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Description),
+		Gateway:      types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].Gateway),
+		PrefixLength: types.Int64Value(int64(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].PrefixLength)),
+		PrimaryDNS:   types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].DNSServer1),
+		SecondaryDNS: types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].DNSServer2),
+		SuffixDNS:    types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].DNSSuffix),
+	}
+
+	// Get network static IP pools
+	ipPools := []staticIPPoolResourceModel{}
+	if len(orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].IPRanges.Values) > 0 {
+		for _, ipRange := range orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values[0].IPRanges.Values {
+			ipPools = append(ipPools, staticIPPoolResourceModel{
+				StartAddress: types.StringValue(ipRange.StartAddress),
+				EndAddress:   types.StringValue(ipRange.EndAddress),
+			})
+		}
+	}
+	var diags diag.Diagnostics
+	plan.StaticIPPool, diags = types.SetValueFrom(ctx, types.ObjectType{AttrTypes: staticIPPoolResourceModelAttrTypes}, ipPools)
+	resp.Diagnostics.Append(diags...)
+
+	// Set refreshed state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *networkIsolatedResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var (
+		plan, state *networkIsolatedResourceModel
+		org         *govcd.Org
+		err         error
+		orgNetwork  *govcd.OpenApiOrgVdcNetwork
+	)
+
+	// Get current state
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Org and VDC
+	org, _, err = r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("[UPDATE] Error retrieving Org", err.Error())
+		return
+	}
```

Comment:
here if VDC is not define globally the `r.client.GetOrgAndVDC` Func return error

Reply:
Yes this problem is true for every resources we've coded.